tests/appsec/iast/sink/test_template_injection.py

```diff
@@ -0,0 +1,66 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+from utils import features, context
+from ..utils import BaseSinkTest
+
+
+@features.iast_sink_template_injection
+class TestTemplateInjection(BaseSinkTest):
+    """Test template injection detection."""
+
+    vulnerability_type = "TEMPLATE_INJECTION"
+    http_method = "POST"
+    insecure_endpoint = "/iast/template_injection/test_insecure"
+    secure_endpoint = "/iast/template_injection/test_secure"
+    location_map = {
+        "nodejs": {"express4": "iast/index.js", "express4-typescript": "iast.ts"},
+    }
+
+    library_configs = {"nodejs": {"pug": "p Hello!", "handlebars": "<p>Hello!</p>"}}
+
+    def setup_insecure(self):
+        configs = self.library_configs.get(context.library.library, {})
+        if configs:
+            for library, template in configs.items():
+                self.data = {"library": library, "template": template}
+                super().setup_insecure()
```

Comment:
This call to `setup_insecure` is only doing one request (the first one). See [this](https://github.com/DataDog/system-tests/blob/main/tests/appsec/iast/utils.py#L110), it is checking if there is an `insecure_request` already attached to the class object before doing a new request.

It happens the same for `setup_secure`.

Reply:
Done